src/Symfony/Component/Form/Extension/Core/Type/TimeType.php

```diff
@@ -210,6 +210,20 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 new DateTimeToArrayTransformer($options['model_timezone'], $options['model_timezone'], $parts, 'text' === $options['widget'], $options['reference_date'])
             ));
         }
+
+        if (\in_array($options['input'], ['datetime', 'datetime_immutable'], true) && null !== $options['model_timezone']) {
+            $builder->addEventListener(FormEvents::PRE_SET_DATA, static function (FormEvent $event) use ($options): void {
```

Comment:
This event allow mutating the data, I think we should listen to `POST_SET_DATA` instead.
Unless we have to hook before the transformation process, in such case I would just use a (very) low priority.

Reply:
Ok, when some initial data is set there are cases no model transformation happens, so we could listen to `POST_SET_DATA` indeed.